Support optrace

limintang opened the pull request:
Differential Revision: D64941627

cccclai commented:
@chiwwang Limin added support for for op trace? Mind taking a look?

shewu-quic commented on backends/qualcomm/runtime/backends/QnnProfiler.cpp:
```diff
@@ -20,7 +20,7 @@ QnnProfile::QnnProfile(
   if (profile_level != QnnExecuTorchProfileLevel::kProfileOff) {
     const QnnInterface& qnn_interface = implementation_.GetQnnInterface();
     Qnn_ErrorHandle_t error = qnn_interface.qnn_profile_create(
-        backend_->GetHandle(), static_cast<int>(profile_level), &handle_);
```
Do you want to fixed profile_level?

chiwwang replied:
maybe we should add the level to `QnnExecuTorchProfileLevel`. Maybe typo? I saw `kProfileOptrace = 3` there.

@shewu-quic , how about the linting profile level? Should we reserve a number for that?

limintang replied:
Sorry, somehow the code change exported is out of sync with the internal diff. Just re-exported; the profile level should be 2 for both kProfileDetailed (2) and kProfileOptrace (3).

limintang replied:
Updated to remove the hardcoded value of QNN profile level.

chiwwang commented on backends/qualcomm/utils/utils.py:
```diff
@@ -770,6 +770,7 @@ def generate_qnn_executorch_compiler_spec(
     online_prepare: bool = False,
     dump_intermediate_outputs: bool = False,
     profile: bool = False,
+    optrace: bool = False,
```
How about change `profile: bool` to an enum class or something similar? Then users can choose profile level they want.

limintang replied:
Good point. However, before this change kProfileDetailed is the only supported profile level. It might be better to make this change in a separate diff.

chiwwang reviewed (commented):
Thank your for the change!

Generally looks good for me. Just some comments around the `profile_level` settings.

limintang committed:
Support optrace (#6535)

Summary:
Add a new profile level to support optrace capture.


Reviewed By: billmguo

Differential Revision: D64941627

shewu-quic reviewed (approved):
LGTM. Thanks for your effort.

chiwwang commented:
Looks good for me. Thank you!

cccclai reviewed (approved):
Thanks folk for the review!

limintang reviewed (commented):
@pytorchbot merge
